DataService.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  DataService.swift
+//  DevAcademy
+//
+//  Created by Zdenko Čepan on 18/07/2023.
+//
+
+import Foundation
+
+enum MyError: Error {
+    case noData
+}
+
+
+class DataService {
+    var data: Result<Features, Error>?
+    
+    static let shared = DataService.init()
+    
+    private init() {
+        
+    }
+    
+    func fetchData(_ n: @escaping (Result<Features, Error>) -> Void) {
+        guard data == nil else {
+            n(Result.failure(MyError.noData))
```

Comment:
v pripade, ze mame data, tak se nejedna o chybu, ale chceme vratit ta data (opet se omlouvam za zmateni na workshopu, ten MyError v ramci tohoto cviceni nakonec nevyuzijeme 😉).

Proto bych mozna namisto `guard` volil `if` a mohlo by to vypadat nejak takhle:
```Swift
if let data = data {
    n(data)
    return
}
```